### ISSUE TITLE
JSApplicationIllegalArgumentException: Error while updating property 'alignItems' in shadow node of type: RCTView

### ISSUE THREAD

**Issue by AliSoftware** [wordpress-mobile/WordPress-Android#16362]
**Sentry Url**: https://sentry.io/share/issue/0ecb07ab6b8d459e8b00fa099d87dfe9/
**User Count**: 187
**Count**: 208
**First Release**: 19.5-rc-2
**First Seen**: 2022-03-29T19:14:34.427000Z
**Last Seen**: 2022-04-19T12:44:10Z
**24 Hours**: 18
**30 Days**: 208

```
JSApplicationIllegalArgumentException: Error while updating property 'alignItems' in shadow node of type: RCTView

...
(8 additional frame(s) were not displayed)
```

**Comment by fluiddot** [comment]
I've analyzed several events related to this issue but I couldn't find a reliable pattern that leads to any clue about the culprit. 

In a couple of cases, I saw two `editor_session_start` events triggered at the same time, which makes me think if this might be related to a case where users open two instances of the editor. In fact, I saw [some comments](https://github.com/wordpress-mobile/WordPress-Android/issues/10493#issuecomment-638176907) about something similar in older issues also referencing the error `JSApplicationIllegalArgumentException: Error while updating property X in shadow node of type: RCTView` 🤔. In any case, I couldn't validate this theory as I'm unable to have two instances of the editor (I tried tapping very quickly on a post but doesn't open two instances).

I also tried using the post's content extracted from some of the user sessions, but still, I couldn't find a particular block that might be causing the error. I also tried using different themes but similarly, doesn't look related.

Additionally, per the Sentry events, I noticed that it's not unique to a property (in this case `alignItems`) but several of them, so it makes the issue even more random to be reproduced.

<img width="500" alt="Screenshot 2022-04-20 at 17 45 21" src="https://user-images.githubusercontent.com/14905380/164473253-7bb35d26-c635-4b2b-bb35-851b2f19d515.png">

<hr>

I ran out of ideas in order to narrow down the cause, so, I'd appreciate it if someone else could take a look in case I'm overlooking anything, thanks 🙇 .

cc @geriux as has been involved in similar issues that are already addressed. I'm wondering if this might be another case related to themes and colors 🤔 , wdyt?

**Comment by antonis** [comment]
Thank you for investigating this @fluiddot 🙇 

> I noticed that it's not unique to a property (in this case alignItems) but several of them, so it makes the issue even more random to be reproduced.

Indeed there are [many occurrences of this exception](https://sentry.io/organizations/a8c/issues/?project=1438088&query=is%3Aunresolved+JSApplicationIllegalArgumentException&sort=freq) 🤔

**Comment by geriux** [comment]
> cc @geriux as has been involved in similar issues that are already addressed. I'm wondering if this might be another case related to themes and colors 🤔 , wdyt?

I'll check it out to see if I find something, I'll report back 👍

**Comment by geriux** [comment]
>cc @geriux as has been involved in similar issues that are already addressed. I'm wondering if this might be another case related to themes and colors 🤔 , wdyt?

I don't think this one is related to the themes/colors.

I think this is more related to what you've mentioned about having more than one instance of the editor. I wonder though, why this is happening since `19.5`.

I investigated a bit and it looks like this was [addressed](https://github.com/facebook/react-native/pull/32167) in the React native core codebase. And it's available in [0.68](https://github.com/facebook/react-native/blob/main/CHANGELOG.md#v0680).

**Comment by fluiddot** [comment]
> I think this is more related to what you've mentioned about having more than one instance of the editor. I wonder though, why this is happening since `19.5`.

Yep, I wonder the same 🤔.

> I investigated a bit and it looks like this was [addressed](https://github.com/facebook/react-native/pull/32167) in the React native core codebase. And it's available in [0.68](https://github.com/facebook/react-native/blob/main/CHANGELOG.md#v0680).

Good catch! Per the issue's description, it looks quite related to what we're experiencing. Let's see when we upgrade the React Native version if it no longer happens 🤞.

**Comment by ovitrif** [comment]
Seems we're still on RN `0.66.2` in Gutenberg, thus my investigation into this issue stops here given that we're looking forward to see if an upgrade to React Native `0.68` or higher will fix this issue.

Currently based on my understanding the React Native upgrade is on hold, with other work being prioritized higher (internal ref: p9ugOq-34s-p2#comment-6225).

**Comment by geriux** [comment]
>Currently based on my understanding the React Native upgrade is on hold, with other work being prioritized higher

That's correct, hopefully we'll continue with the upgrade in the upcoming days 👍

**Comment by thehenrybyrd** [comment]
@geriux I see React Native was upgraded in [this PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5193), do we expect the upgrade to fix the `alignItems` issue here?

**Comment by geriux** [comment]
> @geriux I see React Native was upgraded in [this PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5193), do we expect the upgrade to fix the `alignItems` issue here?

I think it should fix the issue 😃 let's see once it ships if the Sentry issue reports start to go down.

**Comment by jkmassel** [comment]
It's interesting that we haven't seen this one in a while. Regardless, it's Gutenberg Mobile-related and will be addressed by its replacement with GutenbergKit soon.

**Comment by jkmassel** [comment]
Closing this issue — the referenced Sentry issue no longer exists (the share link returns 404), which means this crash hasn't been seen in at least 90 days. If this crash resurfaces, it will appear as a new Sentry issue.
